### PR TITLE
Revert peer dep for types/react-dom which asked for a 16.x version that has not been released

### LINKED ISSUE #30196: [Bug]: react-components peer-dep `@types/react-dom` requires non-existent version for 16.x
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (14) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 3.76 GB / 10.75 GB
  Browsers:
    Chrome: 120.0.6099.130
    Edge: Chromium (120.0.2210.77)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

The source code is obviously wrong without need for a repro

## Actual Behavior

When trying to install `@fluentui/react-components@9.27.4` using `react@16.x`, the installation fails with error such as:
```
│ │ ├─┬ @fluentui/react-aria 9.6.0
│ │ │ ├── ✕ unmet peer react@">=16.14.0 <19.0.0": found 16.13.1
│ │ │ ├── ✕ unmet peer @types/react-dom@">=16.14.0 <19.0.0": found 16.9.14
│ │ │ ├── ✕ unmet peer react-dom@">=16.14.0 <19.0.0": found 16.13.1
```

The latest published version of `@types/react-dom@16.x` is `16.9.24`.

However FluentUI's **package.json** constraint requires `">=16.14.0 <19.0.0"`.  

## Expected Behavior

The mentioned version `16.14.0` is nonsensical.  The `"peerDependencies"` field should refer to version numbers that actually exist.  

Alternatively, if FluentUI has dropped support for React 16, then the version ranges should not refer to version `16`.

### Requested priority

Blocking

### Products/sites affected

https://github.com/microsoft/rushstack/tree/main/vscode-extensions/rush-vscode-command-webview

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by micahgodbolt**
Fixes #30196 was caused by #28959

updated the template file to match the current versions

### CODE REVIEW

File: packages/react-components/global-context/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
   },
   "peerDependencies": {
     "@types/react": ">=16.14.0 <19.0.0",
-    "@types/react-dom": ">=16.14.0 <19.0.0",
+    "@types/react-dom": ">=16.9.0 <19.0.0",
```
Comment (spmonahan) [inline packages/react-components/global-context/package.json:41]: Just curious, why not `>=16.9.24 <19.0.0`?

**Review by spmonahan — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Avatar-mount_base.html">613</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Avatar-mount.html">621</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Button-mount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Button-mount.html">292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Field-mount_base.html">1129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Field-mount.html">1113</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProvider-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProvider-mount.html">715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">61</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">63</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/MakeStyles-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/MakeStyles-mount.html">852</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Persona-mount_base.html">1746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/Persona-mount.html">1756</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/SpinButton-mount_base.html">1394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30259/perf-test-react-components/SpinButton-mount.html">1393</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                                                                    |          `510 B`<br />`328 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr>                                                    |          `537 B`<br />`339 B` |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `93.891 kB`<br />`28.677 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `83.737 kB`<br />`23.474 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `50.175 kB`<br />`15.944 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `19.704 kB`<br />`7.796 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `64.831 kB`<br />`20.274 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `26.905 kB`<br />`8.729 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `27.806 kB`<br />`9.024 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |   `25.311 kB`<br />`9.306 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |  `39.513 kB`<br />`11.169 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `46.874 kB`<br />`12.661 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `44.292 kB`<br />`12.542 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `52.306 kB`<br />`14.134 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `56.559 kB`<br />`13.068 kB` |
| <samp>react-calendar-compat</samp> <br /> <abbr title='bundle-size/Calendar.fixture.js'>Calendar Compat</abbr>                                                                | `142.165 kB`<br />`37.153 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `94.685 kB`<br />`27.248 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `89.509 kB`<br />`25.733 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |   `13.064 kB`<br />`5.385 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |   `15.307 kB`<br />`6.143 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |   `14.015 kB`<br />`5.752 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |  `34.746 kB`<br />`11.902 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `95.176 kB`<br />`30.712 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `93.762 kB`<br />`30.392 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.901 kB`<br />`20.261 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `209.905 kB`<br />`59.925 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `42.388 kB`<br />`14.103 kB` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr>                                                          | `213.658 kB`<br />`59.949 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `95.452 kB`<br />`28.456 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |    `20.82 kB`<br />`7.763 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                                                                       |   `22.427 kB`<br />`8.573 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |   `15.736 kB`<br />`6.228 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `132.737 kB`<br />`41.771 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>                                                                          | `136.424 kB`<br />`42.903 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `26.887 kB`<br />`9.118 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Classic.fixture.js'>Classic Pragma</abbr>                                                                      |       `1.057 kB`<br />`530 B` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/DevRuntime.fixture.js'>JSX Dev Runtime</abbr>                                                                  |    `3.783 kB`<br />`1.648 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Runtime.fixture.js'>JSX Runtime</abbr>                                                                         |    `4.377 kB`<br />`1.881 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |   `14.149 kB`<br />`5.759 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `17.082 kB`<br />`6.911 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          |  `142.09 kB`<br />`43.583 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `144.776 kB`<br />`44.092 kB` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr>                                                         |   `25.181 kB`<br />`9.039 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `12.837 kB`<br />`4.818 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |  `57.066 kB`<br />`17.821 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `120.931 kB`<br />`38.165 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |    `12.788 kB`<br />`4.67 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `7.099 kB`<br />`2.385 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |   `25.891 kB`<br />`9.362 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |   `17.428 kB`<br />`6.898 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `22.928 kB`<br />`8.472 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `32.042 kB`<br />`10.046 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `15.345 kB`<br />`6.263 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |  `28.609 kB`<br />`10.205 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `39.15 kB`<br />`12.769 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `36.774 kB`<br />`11.788 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `23.402 kB`<br />`8.467 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `34.23 kB`<br />`10.995 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `156.347 kB`<br />`43.735 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.899 kB`<br />`13.781 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 |     `129.567 kB`<br />`35 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `74.744 kB`<br />`20.215 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `73.351 kB`<br />`19.806 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr>                                                                      |   `15.259 kB`<br />`6.058 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                                                                            |   `29.982 kB`<br />`9.439 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>                                                                                  |  `74.334 kB`<br />`22.284 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `16.705 kB`<br />`6.569 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `19.878 kB`<br />`6.896 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |  `30.947 kB`<br />`10.476 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr>                                                                 |  `97.013 kB`<br />`32.163 kB` |
| <samp>react-timepicker-compat-preview</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr>                                                         |  `97.013 kB`<br />`32.163 kB` |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr>                                                                   |  `93.203 kB`<br />`28.026 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `53.371 kB`<br />`18.855 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/99e1128925fa5863a99e311884cc2c710c2606bd'>99e1128925fa5863a99e311884cc2c710c2606bd</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/global-context/package.json (hunk 1 above)
Comment (micahgodbolt) [reply packages/react-components/global-context/package.json:41]: none of the other deps used the last value, so I assumed it was convection. react types tops out at 16.14.55

**Review by khmakoto — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Review by behowell — APPROVED** [review]

**Review by mltejera — APPROVED** [review]

**Review by bsunderhus — APPROVED** [review]

**Review by Hotell — APPROVED** [review]

**Comment by octogonz** [comment]
Thank you! 👍

**Pull request merged**